[yashma-sonara] iP

yashma-sonara opened the pull request:
## Chuklbot

>Chuklbot lets you forget the constant list of tasks in your head and fills it with jokes and laughter. 

It is:

- Text based
- easy to learn
- ~~FAST~~ super fast to use


All you need to do is: 
1. Download it from [here](https://github.com/yashma-sonara/ip)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yashma-sonara committed:
Completed Level-0 increment

yashma-sonara committed:
Completed Level-1 increment

yashma-sonara committed:
Completed Level-2 increment

yashma-sonara committed:
Completed Level-3 increment

yashma-sonara committed:
Completed Level 4 Increment

yashma-sonara committed:
Completed A-TextUiTesting

yashma-sonara committed:
Completed Level-5 minimally

yashma-sonara committed:
Fix Deadline Exceptions

yashma-sonara committed:
Fix Event Exceptions

yashma-sonara committed:
Completed Level-6 Increment

yashma-sonara committed:
Completed A-Enums

yashma-sonara committed:
Level 7: Implement automatic saving and loading of tasks

yashma-sonara committed:
Level 7: Implement automatic saving

yashma-sonara committed:
Update to handle corrupted data file

yashma-sonara committed:
Merge branch-Level-7 into master

yashma-sonara committed:
Update to add date and time for deadline tasks

yashma-sonara committed:
Merge branch-Level-8 into master

yashma-sonara committed:
Update add OOP classes and key functionalities work

yashma-sonara committed:
Revert "Update add OOP classes and key functionalities work"

This reverts commit e878fd4e348cad53303c451b9a6e326cd695a45b.

yashma-sonara committed:
Revert "Revert "Update add OOP classes and key functionalities work""

This reverts commit 65015df8991566f1d7921f59b4e2c1d92f3992ac.

yashma-sonara committed:
Merge branch 'branch-Level-8' into A-MoreOOP

yashma-sonara committed:
Update delete task and todo task

yashma-sonara committed:
update formatting

yashma-sonara committed:
Merge branch 'A-MoreOOP'

yashma-sonara committed:
update mark formating

yashma-sonara committed:
add duke package

yashma-sonara committed:
add task package

yashma-sonara committed:
Merge branch 'branch-Packages'

yashma-sonara committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yashma-sonara committed:
add todo unit test

yashma-sonara committed:
Add unit tests for tasks, ui, and tasklist

yashma-sonara committed:
Merge remote-tracking branch 'origin/A-JUnit'

yashma-sonara committed:
add javadocs for task package

yashma-sonara committed:
add javadocs for remaining classes

yashma-sonara committed:
add javadocs for test classes

yashma-sonara committed:
add find capability

yashma-sonara committed:
Merge branch 'branch-A-JavaDoc'

yashma-sonara committed:
update some formatting

yashma-sonara committed:
fix merge issues

yashma-sonara committed:
Merge branch 'branch-Level-9'

yashma-sonara committed:
add checkstyle setup

yashma-sonara committed:
update enum

yashma-sonara committed:
update checkstyle

yashma-sonara committed:
complete checkstyle

yashma-sonara committed:
Merge branch 'A-CheckStyle' into master

yashma-sonara committed:
complete tutorial part 2

yashma-sonara committed:
complete base gui

yashma-sonara committed:
complete base gui + some styling

yashma-sonara committed:
complete basic gui functions

yashma-sonara committed:
update multiple word inputs for task + list format

yashma-sonara committed:
compete gui

yashma-sonara committed:
Merge branch-Level-10 into master

Rikko1204 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+     * @param file The file path to save and load tasks.
+     */
+    public Storage(String file) {
+        this.File_Path = file;
```
Just a little note there is no need for 'this' as the names are distinguishable 😄

Rikko1204 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+ * Storage class handles the loading and saving of tasks to a file.
+ */
+public class Storage {
+    private final String File_Path;
```
Remember that fields marked with final should all be in block letters!

Rikko1204 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,157 @@
+     *
+     * @return The user's input command as a String.
+     */
+    public String getUserInput3() {
+        //System.out.print("Enter your command: ");
+        return scanner.nextLine();
+    }
+
+    /**
+     * Displays a message indicating a deleted command.
+     *
+     * @param str The command that was deleted.
+     */
+    public String deleteMessage(String str) {
+        String res = "I have deleted the below command: \n" + str;
+        System.out.println(res);
+        return res;
+    }
+
+    /**
+     * Gets user input from the command line, expecting a single word.
+     *
+     * @return The user's input command as a String.
+     */
+    public String getUserInput2() {
+        //System.out.print("Enter your command: ");
+        return scanner.next();
+    }
+
+    /**
+     * Gets user input as an integer from the command line.
+     *
+     * @return The user's input as an integer.
+     */
+    public int getUserInputInt() {
+        return scanner.nextInt();
+    }
```
Perhaps you could find a way to combine all getUserInput into one method?

Rikko1204 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,100 @@
+ * Duke is a task management application that allows users to manage their tasks.
+ * It provides a command-line interface and a JavaFX GUI for users to interact with their task list.
+ */
+public class Duke extends Application {
```
Perhaps consider giving your chatbot a unique name and personality? Would be fun to play around 😆

Rikko1204 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,55 @@
+ * Represents exceptions specific to the Duke program.
+ */
+
+public class DukeException extends Exception {
```
I like how the exceptions have been broken down into subclasses

Rikko1204 reviewed (commented):
LGTM! Just a few nits to fix, keep up the good work! 👍

yashma-sonara committed:
add some assertions in parser and storage

yashma-sonara committed:
add some more assertions

yashma-sonara committed:
refactor some code to remove warnings

yashma-sonara committed:
refactor remaining code

yashma-sonara committed:
Merge pull request #1 from yashma-sonara/branch-A-Assertions

Integrate assertions in codebase

yashma-sonara committed:
Merge branch 'master' into branch-A-CodeQuality

yashma-sonara committed:
Merge pull request #2 from yashma-sonara/branch-A-CodeQuality

Refactor codebase

yashma-sonara committed:
Add CI/CD config

yashma-sonara committed:
Add C-DetectDuplicates feature

Previously, the application did not handle duplicate items, which could lead to inconsistencies in task management.

Handling duplicate items is essential for maintaining data integrity, preventing unintended actions,

Lets,
* allow parser to detect duplicate tasks and prevent adding of the task
* report to user that duplicate task was entered

The approach chosen ensures that only unique tasks are stored in the task list, reducing clutter and minimizing confusion for users.

myang2020 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,68 @@
+        }
+
+        dialog.setText(text);
+        Circle circle = new Circle(40, 40, 30);
```
Perhaps have a more descriptive name than "circle".

myang2020 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,80 @@
+     * @param stage The primary stage for this application.
+     */
+    @Override
+    public void start(Stage stage) {
```
Perhaps specify what is going to start.

myang2020 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,55 @@
+     * Exception class for an empty description in a Todo task.
+     * This exception is thrown when attempting to create a Todo task with an empty description.
+     */
+    public static class EmptyTodoDescriptionException extends DukeException {
```
Great job following PascalCase.

myang2020 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,30 @@
+ */
+public class Main extends Application {
+
+    private Duke duke = new Duke("data/duke.txt");
```
Perhaps give your chatbot a unique name!

myang2020 reviewed (commented):
LGTM! Just a few variable/function names that could be more descriptive. Keep up the good work and good luck ahead!

yashma-sonara committed:
Unify variations of toSet() methods

yashma-sonara committed:
Update redundancies

yashma-sonara committed:
Refactor storage class: extract file parsing

Parser class is handling stored file parsing.

Extracting the parseTaskFromLine method allows us to encapsulate file
methods in the storage class.

Let's add a file parsing method to the storage class.

yashma-sonara committed:
Fix checkstyle

yashma-sonara committed:
Change name of chatbot and update to gui

yashma-sonara committed:
Merge branch-A-BetterGui: Added gui update

yashma-sonara committed:
Update README.md

yashma-sonara committed:
Update gui images

yashma-sonara committed:
Merge branch 'master' of https://github.com/yashma-sonara/ip

yashma-sonara committed:
Update gui

yashma-sonara committed:
Add Ui image

yashma-sonara committed:
Update README.md

yashma-sonara committed:
Update exceptions

yashma-sonara committed:
Merge branch 'master' of https://github.com/yashma-sonara/ip

yashma-sonara committed:
Add closing for bye command

yashma-sonara committed:
Add Joy's response details

yashma-sonara committed:
Update ui picture

yashma-sonara committed:
Update README.md

yashma-sonara committed:
Fix deadline time issue

yashma-sonara committed:
Merge branch 'master' of https://github.com/yashma-sonara/ip

yashma-sonara committed:
Update README.md

yashma-sonara committed:
fix bug

yashma-sonara committed:
Merge branch 'master' of https://github.com/yashma-sonara/ip

yashma-sonara committed:
Fix checkstyle
